compiler-rt/src/integer/u40.cairo

```diff
@@ -0,0 +1,70 @@
+use crate::alu::and::and_i64;
+use crate::alu::and::and_i128;
+use core::num::traits::{BitSize, Bounded};
+
+#[derive(Drop)]
+pub struct u40 {
+    // The 40bits integer is represented as a 128bits unsigned int.
+    data: u128,
+}
+
+fn u40_new(value: u64) -> u40 {
+    let bitmask: u128 = 0b1111111111111111111111111111111111111111; // 1<<40
+    let significant_bits = and_i64::__llvm_and_l_l_l(value.into(), bitmask);
+    if significant_bits != value.into() {
+        panic!("value = {} does not fit in 40bits", value)
+    }
+    u40 { data: significant_bits }
+}
+
+impl U40BitSize of BitSize<u40> {
+    fn bits() -> usize {
+        40
+    }
+}
+
+impl I40IntoU128 of Into<u40, u128> {
+    fn into(self: u40) -> u128 {
+        return self.data;
+    }
+}
+
+impl BoundedU40 of Bounded<u40> {
+    const MIN: u40 = u40 { data: 0b0 };
+    const MAX: u40 = u40 { data: 0b1111111111111111111111111111111111111111 }; // 1<<40
```

Comment:
This is redefined from above. Please define a separate constant and use it in both places.